### PR TITLE
Exclude from MinSeq rather than Kickout Idle Clients

### PR DESCRIPTION
**Pull request by anthony-murphy**
The behavior here was to issues a leave message for idle clients to kick them out. The reason this was done was to prevent them from holding back the minimum sequence number and therefore extending the collaboration window. However, this behavior has drawbacks, as it is difficult to craft an accurate heuristic for when to kick client outs, and maintains the desired behavior of keeping the collaboration window as small as possible, and only kicking out client that are truly holding the collaboration window back.

This change makes it such that we no longer kick clients out of the session. Instead, if we detect that a client is idle and holding back the collaboration window we just exclude them from calculating the window, specifically the minimum sequence number.  When they send their next op it will either be in the window or not. If not the op will be NACK'd and the client will disconnect itself. If the op is in the window the client's ref seq will be reset and the client can continue collaborating without needing to reconnect.

### CODE REVIEW

**Review by tanviraumi — COMMENTED** [review]
The refactoring part looks good. Can we please split this PR into two. It's very hard to follow the actual change.

Also I am not sure if I am understanding this correctly. What is the issue with actively sending leave message? Is there a bug somewhere?

**Review by vladsud — APPROVED** [review]
:shipit:
